broadinstitute/pearl
# [JN-463] Support adding new question templates

This adds the ability to A) create new question templates and B) create new questions that use those question templates (see videos below)

TO TEST:

1. Try creating new question templates of different question types
2. Try creating new questions based on those templates. Make sure that the optional overrides work correctly
3. When creating a new question, try switching back and forth between using a template and not using a template. Make sure that the appropriate fields are disabled when a question template is selected, and that when saving a question afterwards, the correct fields are saved in the JSON
4. Verify that the questions show up correctly in the survey preview

UPDATE: there is one bug that I noticed after requesting reviews. If you select a question type, and then select a question template, it doesn't re-render the question type dropdown after setting the type to `undefined`. Not sure why this is happening. The underlying state is correct, so it's mainly a rendering problem. Looking into it...

Adding a new question template:


https://github.com/broadinstitute/juniper/assets/7257391/2ecfcba7-95c4-4f85-8b77-9d5f440ecae0



Using a question template to construct a new question:

https://github.com/broadinstitute/juniper/assets/7257391/c7748c0c-c125-4307-93e3-904860249e83
